cfpb/cfgov-refresh
# Move Blog, Newsroom, Activity log to Wagtail

I've combined this into a single PR because you need each piece to test it all without breaking things.

## Additions

- Legacy page types to migrate old content from WordPress
- import-data processors to do the migrations
- corresponding templates
- rss feeds
- new references to Activity log in the category reference
- Reports is now a thing in Activity Log

## Removals

- i removed the old stuff that we aren't going to use anymore, like urls and template files

## Changes

- The `get_page_set` function to get the pages for each of the filterable lists ended up being so different that i ended up breaking into its corresponding page class to separate concerns.
- legacy content uses the same article.html file it did before but i changed it to be wagtail page compatible.
- view more works with wagtail pages now
- bug fixing some stuff
- some other stuff i probably forgot

## Testing

- get the latest from refresh
- make sure you have the database option in local.py set to run migrations that create foreign key references
- run the migrations
- under the home page, add an Activity Log Page
 - make sure there's a filterable list with categories selected, and **Activity Log** selected as the page type
 - ![screen shot 2016-04-19 at 10 19 47 am](https://cloud.githubusercontent.com/assets/254877/14642382/46b8d764-0618-11e6-8378-28de8ec78f41.png)

- under the about us section, add a **Newsroom Landing Page** and a Blog **Sublanding filterable Page** 
 - again, make sure that there's a filterable list, and the page types are selected (Newsroom for Newsroom, Blog for blog)
- run `./cfgov/manage.py import_data posts legacyblogpage --parent blog -u user -p pass`
- run `./cfgov/manage.py import_data newsroom legacynewsroompage --parent newsroom -u user -p pass`
 - *Note*: a couple errors are fine, just let me know what they are. there are some blog posts that should exist in production right now
- Play around with the list filters on each page to make sure that they work. things like filtering on date, topic, multiple categories, and the like.
- Next, go to a Blog post and make sure that the further reading organism is pulling content if it exists for that tag, and that the View More link goes to the activity log with the appropriate tags selected.
- I think that's it....

## Review

- @rosskarchner 
- @kave 
- @richaagarwal 

## Screenshots
Blog Landing Page
![screen shot 2016-04-18 at 7 20 09 pm](https://cloud.githubusercontent.com/assets/1412978/14622796/972bad3a-059a-11e6-8d40-c34f14955268.png)


LegacyBlogPage
![screen shot 2016-04-18 at 7 16 07 pm](https://cloud.githubusercontent.com/assets/1412978/14622711/0c3260e8-059a-11e6-9227-99f82040c5a1.png)

Newsroom Landing Page (note the side nav. this is pulling in all browse type pages but newsroom is currently not doing this. its a flag in the admin to set it right. Also, theres a child page of Newsroom called press sources that isn't there; more below.)
![screen shot 2016-04-18 at 7 17 13 pm](https://cloud.githubusercontent.com/assets/1412978/14622781/83ec44e6-059a-11e6-9cdd-0576867cf02e.png)


LegacyNewsroomPage (same template as blog but this doesn't have further reading content for lack of tags)
![screen shot 2016-04-18 at 7 17 48 pm](https://cloud.githubusercontent.com/assets/1412978/14622749/407902b2-059a-11e6-8ff0-2425074a26d5.png)

Activity Log pulling in reports alongside blog
![screen shot 2016-04-18 at 7 19 00 pm](https://cloud.githubusercontent.com/assets/1412978/14622771/75429b5c-059a-11e6-95c1-40a9a5672460.png)


## Notes

- @schaferjh Can the Press Sources page (under Newsroom) be a Wagtail page right now?

## Todos

- We still need to make the new BlogPage and the new NewsroomPage classes and templates
- Possibly create the Press Sources page
